src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -133,9 +133,9 @@ class RegularSync(
   def handleNewBlockHashesMessages: Receive = {
     case MessageFromPeer(NewBlockHashes(hashes), peerId) =>
       val maybePeer = peersToDownloadFrom.find(peer => peer._1.id == peerId)
-      //we allow asking for new hashes when we are not syncing and we can download from specified peer and we are
-      //top of the chain
-      if (notDownloading() && topOfTheChain && maybePeer.isDefined) {
+      //we allow asking for new hashes when we are not syncing and we can download from specified peer ,we are
+      //top of the chain and not resolving branches currently
+      if (notDownloading() && topOfTheChain && maybePeer.isDefined && !resolvingBranches) {
```

Comment:
Do we maybe need to add this check also for handling the `NewBlock` messages? Maybe this check could then be added to the `notDownloading()` function?